[Beta 15.5.1] Modifie l'ajout de la barre de recherche

Luuka opened the pull request:
| Q | R |
| --- | --- |
| Correction de bugs ? | oui |
| Nouvelle Fonctionnalité ? | non |
| Tickets (_issues_) concernés | #2679 |

La barre de recherche du menu mobile était mal ajoutée sur la page d'accueil
### QA
- Il faut au préalable avoir builder le front : `npm run gulp -- build`
- Ouvrez la page d'accueil ainsi qu'une autre page (tutoriel, forum...)
- Diminuez la taille de la fenêtre jusqu'à ce que l'ouverture du menu mobile soit disponible.
- Vérifiez que la barre de recherche en haut du menu mobile soit identique sur les 2 pages.

NB : C'est ma première contribution reprenez moi si j'ai fait une boulette dans la façon de faire :)

Luuka committed:
Ref #2679 : modifie l'ajout de la barre de recherche

Luuka committed:
Ref #2679 : modifie l'ajout de la barre de recherche

Situphen commented:
Je suis pas fan de l'HTML à l'intérieur du JS mais je vois pas trop d'autres solutions. Enfin, on pourrait changer de place le `id="search"` de `templates/home.html` mais on aurait sûrement un rendu différent.

@sandhose : un avis ?

Situphen commented:
> NB : C'est ma première contribution reprenez moi si j'ai fait une boulette dans la façon de faire :)

Ça m'a l'air parfait ! ;)

Luuka commented:
Je suis d'accord sur le coté un peu crado.

Une autre solution pourrait être de remettre le fil d'Ariane dans le DOM mais de le cacher en CSS, comme ça on garde le JS.

> Ça m'a l'air parfait ! ;)

Merki ! :)

Situphen commented:
On était parti sur cette solution au début.

GerardPaligot committed:
feat(home): Reduces hits on database.

Adds select_related and prefetch_related methods on
queries used by the home view.

Eskimon commented:
Luuka tu pourrais `rebase` ta branche sur celle de release afin que Travis fonctionne ? ( https://zestedesavoir.com/tutoriels/649/refaire-lhistoire-avec-git/#4-merger-une-autre-branche )

Eskimon commented:
Tout marche proprement, une fois le rebase fait c'est bon pour merge :)

SpaceFox committed:
Merge pull request #2701 from GerardPaligot/fix_less_request

Reduces hits on database.

Situphen commented:
J'aimerais quand même l'avis de @sandhose sur la solution adaptée. Peut-être qu'il en a une meilleure !

pierre-24 committed:
Merge pull request #2677 from SpaceFox/fix_2640_requirements

[Beta 15.5.1] MAJ des dépendances

Luuka committed:
Merge branch 'fix-2679' of github.com:Luuka/zds-site into fix-2679

sandhose commented:
Ca marche, ça fixe le bug, _mais_ il y a plusieurs problèmes avec cette solution.

En plus du fait que c'est dégueu de générer du DOM comme ça dans le JS, ca prends pas du tout en compte l'internationalisation.

Je préférerais garder le `subheader`, et mettre un `display: none` dessus.

Donc, je ferme, et si tu veux, tu peux faire une PR qui remets le subheader dans le template de la home, et mettre un `display: none` dessus :)

Situphen commented:
Il faut faire le contraire de ce _commit_ : https://github.com/zestedesavoir/zds-site/commit/259a685bbe2064bffbbb91e12d72b2e4a5dcfdf4

Sauf pour le `{% block pre_content %}`, il ne faut pas le remettre !
